src/System.Management.Automation/FormatAndOutput/common/StringDecorated.cs

```diff
@@ -109,8 +109,11 @@ private string PlainText
         // CSI escape sequences
         private const string CsiRegex = @"(\x1b\[\?\d+[hl])";
 
+        // Hyperlink escape sequences. Note: '.*?' makes '.*' do non-greedy match.
+        private const string HyperlinkRegex = @"(\x1b\]8;;.*?\x1b\\)";
```

Comment:
Can we make this more specific instead of using `.*?` which accepts any characters up to an escape?

Reply:
A hyperlink sequence looks like this: ``"`e]8;;link-url-string`e\link-description-string`e]8;;`e\"``.

Good catch. This regex will remove the whole `link-url-string` which could contain arbitrary characters, such as `//`, `#`, or any other comment indicators in other languages.

I thought about doing validation on the `link-url-string` part and only treat it as hyperlink sequence if the `link-url-string` part is valid http/https urls. But that doesn't work because valid url can have `#` in it, which is the comment indicator for any scripting languages.

I think we need to fix `Out-File` to not change anything if the piped in object is text. It should do the VT sequence striping only when it has to go through the formatting system -- meaning only strip the VT sequences that are added during formatting.

Reply:
Opened issue https://github.com/PowerShell/PowerShell/issues/17452, and also here is the PR to fix `Out-File` and `Out-String`: https://github.com/PowerShell/PowerShell/pull/17455. The proposed behavior for `Out-File` and `Out-String` is "Leave ANSI out of object formatting, but leave strings alone".

Reply:
Yes, I think this is resolved